### ISSUE TITLE
Bug: Error message of verify mode is returned as normal text regardless of output mode

### ISSUE THREAD

**Issue by foresterre** [foresterre/cargo-msrv#263]
Command:

```
cargo msrv --verify --output-format json --ignore-lockfile
```

Output:

```
{"reason":"fetching-index"}
{"reason":"mode","mode":"verify-msrv","toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"installing","version":"1.36.0","step":0,"total":1,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"checking","version":"1.36.0","step":0,"total":1,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"check-complete","version":"1.36.0","step":0,"total_steps":1,"success":false,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"verify-complete","success":false,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
Crate source was found to be incompatible with its MSRV '1.36.0', as defined in 'C:\bisector\Cargo.toml'
```

The last line is not json, even though the `--output-format json` option was given.

Present at least since cargo-msrv 0.13.0

**Comment by foresterre** [comment]
Cause: https://github.com/foresterre/cargo-msrv/blob/main/src/bin/cargo-msrv.rs#L17

An error shouldn't be printed directly, instead it should first be passed to the correct output format.

**Comment by foresterre** [comment]
Fixed here: [`5b28b7c` (#357)](https://github.com/foresterre/cargo-msrv/pull/357/commits/5b28b7c07717aeaeec22809b42ba9ed3e077c974#diff-72a8660bff7d91b9c1e7985d434a773cbe58c206ff5f337bcef8fb3ee8bace75L19)
